cpp/src/arrow/sparse_tensor_test.cc

```diff
@@ -903,4 +903,207 @@ TEST_F(TestSparseCSCMatrix, TestToTensor) {
   ASSERT_TRUE(tensor.Equals(*dense_tensor));
 }
 
+template <typename IndexValueType>
+class TestSparseCSFTensorBase : public ::testing::Test {
+ public:
+  void SetUp() {
+    shape_ = {4, 3, 5, 2};
+    dim_names_ = {"a", "b", "c", "d"};
+
+    std::vector<int64_t> dense_values = {
+        0, 1, 2, 0, 0, 0, 0, 0, 0, 0, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 5, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 6, 0, 0, 0, 0, 0, 0, 0, 7, 0, 8};
+    auto dense_data = Buffer::Wrap(dense_values);
+    NumericTensor<Int64Type> dense_tensor(dense_data, shape_, {}, dim_names_);
+    ASSERT_OK_AND_ASSIGN(sparse_tensor_from_dense_,
+                         SparseCSFTensor::Make(
+                             dense_tensor, TypeTraits<IndexValueType>::type_singleton()));
+  }
+
+ protected:
+  std::vector<int64_t> shape_;
+  std::vector<std::string> dim_names_;
+  std::shared_ptr<SparseCSFTensor> sparse_tensor_from_dense_;
+};
+
+class TestSparseCSFTensor : public TestSparseCSFTensorBase<Int64Type> {};
+
+TEST_F(TestSparseCSFTensor, CreationFromTensor) {
+  std::vector<int64_t> values = {
+      0, 1, 2, 0, 0, 0, 0, 0, 0, 0, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+      0, 0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 5, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 6, 0, 0, 0, 0, 0, 0, 0, 7, 0, 8};
+  std::vector<int64_t> shape({4, 3, 5, 2});
+  std::vector<std::string> dim_names({"a", "b", "c", "d"});
+  std::shared_ptr<Buffer> buffer = Buffer::Wrap(values);
+  Tensor tensor(int64(), buffer, shape, {}, dim_names);
+
+  std::shared_ptr<SparseCSFTensor> st;
+  ASSERT_OK_AND_ASSIGN(st, SparseCSFTensor::Make(tensor));
+
+  std::vector<std::vector<int64_t>> indptr_values = {
+      {0, 1, 4, 6}, {0, 1, 2, 3, 4, 5, 6}, {0, 1, 2, 3, 5, 6, 8}};
+  std::vector<std::vector<int64_t>> indices_values = {
+      {1, 0, 1}, {0, 0, 1, 0, 1, 2}, {0, 0, 0, 1, 3, 3}, {0, 1, 0, 0, 3, 4, 3, 4}};
+  std::vector<std::shared_ptr<Buffer>> indptr_buffers(3);
+  std::vector<std::shared_ptr<Buffer>> indices_buffers(4);
+  std::vector<int64_t> axis_order = {3, 1, 0, 2};
+  std::vector<int64_t> indices_shapes = {3, 6, 6, 8};
+
+  for (int64_t i = 0; i < static_cast<int64_t>(indptr_values.size()); ++i)
```

Comment:
I guess this for statement can be written by the range-based style:

```cxx
std::vector<std::shared_ptr<Buffer>> indptr_buffers;  // NOTE: remove the count argument
...
for (auto& indptr : indptr_values)
  indptr_buffers.push_back(Buffer::Wrap(indptr));
```

The `indices_buffers` below can be initialized by the same style.

Reply:
Done.